src/activity_handlers.cpp

```diff
@@ -2866,12 +2866,31 @@ void activity_handlers::dig_channel_do_turn( player_activity *act, player *p )
 void activity_handlers::dig_finish( player_activity *act, player *p )
 {
     const tripoint &pos = act->placement;
-
-    if( g->m.ter( pos ) == t_pit_shallow ) {
+    bool grave = g->m.ter( pos ) == t_grave ? true : false;
+    if( g->m.ter( pos ) == t_pit_shallow || grave ) {
         g->m.ter_set( pos, t_pit );
     } else {
         g->m.ter_set( pos, t_pit_shallow );
     }
+    if( grave ) {
+        if( one_in( 5 ) ) {
```

Comment:
20 percent chance? I'd say that's way too much. I'd lower it it 5 percents or even less, to make it more of a surprise rather than frequent event.

Reply:
I'll consult this and get back to it.

Reply:
I've dropped the chance to 10% (1 in 10 graves average).